### ISSUE TITLE
IReactorTime is missing method to determine the current time

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2634]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2634|
|Type|enhancement|
|Created|2007-05-08 19:47:43Z|

With only a method to schedule a call, code which wants to test time-related functionality has to cobble together an IReactorTimer provider with some other object which can report the current time.  Likewise, implementations have no uniform way to allow their scheduling code to be completely overridden, since it is common for scheduling code needs to know the current time (and this code currently uses ```twisted.python.runtime.seconds``` if it is correct, or ```time.time``` if it is incorrect).

<details><summary>Searchable metadata</summary>

```
trac-id__2634 2634
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1178653663000000 1178653663000000
changetime__1179103111000000 1179103111000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

See [#2633](https://github.com/twisted/twisted/issues/2633) for a branch that should resolve this ticket.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix set status to `closed`|
|-|-|

(In [20273]) Merge parameterize-looping-call-2633.

Author: radix
Reviewer: therve
Fixes [#2633](https://github.com/twisted/twisted/issues/2633)
Fixes [#2634](https://github.com/twisted/twisted/issues/2634)

LoopingCall (and code which uses it) is now much more testable because the
scheduler it uses is overridable. In support of this, a 'seconds' method
was added to the reactor (IReactorTime) to strengthen the logical grouping
of 'what is the current scheduling-time' and 'schedule a call for later'.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
